### PR TITLE
RM6923 support i2c in linux on imxrt1170

### PR DESCRIPTION
**Pull request by asanochkin-emcraft**
**Unit test**
Build the rootfs project with these changes applied.

Unit test performed and is a success.

### CODE REVIEW

File: arch/arm/boot/dts/imxrt1170.dtsi [hunk 1]
```diff
@@ -361,6 +361,57 @@
 			clocks = <&clks IMXRT1170_CLK_USB>;
 		};
 
+		lpi2c1: lpci2c@40104000 {
+			compatible = "fsl,imx7ulp-lpi2c";
+			reg = <0x40104000 0x4000>;
+			interrupts = <32>;
+			/*clocks = <&clks IMXRT1170_CLK_LPI2C1>, <&clks IMXRT1170_CLK_IPG_PODF>;
```
Comment (vdskvortsov) [inline arch/arm/boot/dts/imxrt1170.dtsi]: Please remove the unused code.

File: arch/arm/boot/dts/imxrt1170.dtsi [hunk 2]
```diff
@@ -361,6 +361,57 @@
 			clocks = <&clks IMXRT1170_CLK_USB>;
 		};
 
+		lpi2c1: lpci2c@40104000 {
+			compatible = "fsl,imx7ulp-lpi2c";
```
Comment (vdskvortsov) [inline arch/arm/boot/dts/imxrt1170.dtsi:365]: Shouldn't we also add `address-cells` and `size-cells` here? Please extend unit test to make sure there are not warnings when build DTB.

File: arch/arm/boot/dts/imxrt1170.dtsi [hunk 3]
```diff
@@ -361,6 +361,57 @@
...
+			clocks = <&clks IMXRT1170_CLK_LPI2C1>;
+			clock-names = "per";
+			status = "disabled";
+		};
+
+		lpi2c2: lpci2c@40108000 {
+			compatible = "fsl,imx7ulp-lpi2c";
+			reg = <0x40108000 0x4000>;
+			interrupts = <33>;
+			clocks = <&clks IMXRT1170_CLK_LPI2C2>;
```
Comment (vdskvortsov) [inline arch/arm/boot/dts/imxrt1170.dtsi:378]: Missed `clocks-names` property for the lpi2c[2..6] nodes. If it is not needed, please remove corresponding property in `lpi2c1`.

File: drivers/clk/imx/clk-imxrt1170.c [hunk 4]
```diff
@@ -31,6 +31,8 @@ static const char * const semc_sels[] = {IMXRT1170_CLK_SRC_COMMON,
 "pll1_div5", "pll2_sys", "pll2_pfd1", "pll3_pfd0"};
 static const char * const enet1_sels[] = {IMXRT1170_CLK_SRC_COMMON,
 "pll1_div2", "audio_pll", "pll1_div5", "pll2_pfd1"};
+static const char * const lpi2c1_6_sels[] = {IMXRT1170_CLK_SRC_COMMON,
+"pll3_div2", "pll1_div5", "pll2_pfd3"};
```
Comment (vdskvortsov) [inline drivers/clk/imx/clk-imxrt1170.c]: Must be  `"pll3_div2", "pll1_div5", "pll2_sys", "pll2_pfd3"` Refer to IMXRT1170 RM.

File: drivers/clk/imx/clk-imxrt1170.c [hunk 5]
```diff
@@ -85,6 +87,12 @@ static struct imxrt1170_clk_root clk_roots[] = {
 	{ IMXRT1170_CLK_ROOT_SEMC, "semc_root", semc_sels, (4 * 0x80), CLK_IS_CRITICAL },
 	{ IMXRT1170_CLK_ROOT_GPT1, "gpt1_root", gpt1_sels, (14 * 0x80), },
 	{ IMXRT1170_CLK_ROOT_LPUART1, "lpuart1_root", lpuart1_sels, (25 * 0x80), },
+	{ IMXRT1170_CLK_ROOT_LPI2C1, "lpi2c1_root", lpi2c1_6_sels, (37 * 0x80), },
+	{ IMXRT1170_CLK_ROOT_LPI2C2, "lpi2c2_root", lpi2c1_6_sels, (38 * 0x80), },
+	{ IMXRT1170_CLK_ROOT_LPI2C3, "lpi2c3_root", lpi2c1_6_sels, (39 * 0x80), },
+	{ IMXRT1170_CLK_ROOT_LPI2C4, "lpi2c4_root", lpi2c1_6_sels, (40 * 0x80), },
+	{ IMXRT1170_CLK_ROOT_LPI2C5, "lpi2c5_root", lpi2c1_6_sels, (41 * 0x80), },
```
Comment (vdskvortsov) [inline drivers/clk/imx/clk-imxrt1170.c]: `lpi2c5` and `lpi2c6` root clocks use different mux. Refer to IMXRT1170 RM.

**Review by vdskvortsov — CHANGES_REQUESTED** [review]

File: drivers/clk/imx/clk-imxrt1170.c [hunk 6]
```diff
@@ -31,6 +31,10 @@ static const char * const semc_sels[] = {IMXRT1170_CLK_SRC_COMMON,
 "pll1_div5", "pll2_sys", "pll2_pfd1", "pll3_pfd0"};
 static const char * const enet1_sels[] = {IMXRT1170_CLK_SRC_COMMON,
 "pll1_div2", "audio_pll", "pll1_div5", "pll2_pfd1"};
+static const char * const lpi2c1_4_sels[] = {IMXRT1170_CLK_SRC_COMMON,
+"pll3_div2", "pll1_div5", "pll2_sys", "pll2_pfd3"};
+static const char * const lpi2c5_6_sels[] = {IMXRT1170_CLK_SRC_COMMON,
+"pll3_pfd3", "pll3_clk", "pll2_pfd3", "pll1_div5"};
```
Comment (vdskvortsov) [inline drivers/clk/imx/clk-imxrt1170.c]: There is no `pll3_clk` clock defined in this driver. Did you mean `pll3_sys`?

**Review by vdskvortsov — CHANGES_REQUESTED** [review]

File: drivers/clk/imx/clk-imxrt1170.c (hunk 6 above)
Comment (asanochkin-emcraft) [reply drivers/clk/imx/clk-imxrt1170.c]: Corrected.

File: arch/arm/boot/dts/imxrt1170.dtsi (hunk 1 above)
Comment (asanochkin-emcraft) [reply arch/arm/boot/dts/imxrt1170.dtsi]: Done

File: arch/arm/boot/dts/imxrt1170.dtsi (hunk 2 above)
Comment (asanochkin-emcraft) [reply arch/arm/boot/dts/imxrt1170.dtsi:365]: Moved these nodes from rootfs.dts.IMXRT117X_NXPEVK. Done.

**Review by vdskvortsov — APPROVED** [review]

**Review by vdskvortsov — CHANGES_REQUESTED** [review]
Please made the slave mode in the LPI2C driver configurable. Currently the CONFIG_I2C_SLAVE option is mandatory to build the LPI2C driver, should made it optional.

**Review by vdskvortsov — APPROVED** [review]

**Pull request merged**